.github/scripts/trymerge.py

```diff
@@ -1330,13 +1330,30 @@ def main() -> None:
     org, project = repo.gh_owner_and_name()
     pr = GitHubPR(org, project, args.pr_num)
 
-    def handle_exception(e: Exception, msg: str = "Merge failed") -> None:
-        msg += f"\nReason: {e}"
+    def handle_exception(e: Exception, title: str = "Merge failed") -> None:
+        exception = f"**Reason**: {e}"
+
+        troubleshooting = ""
+        if args.land_checks:
+            troubleshooting += get_land_check_troubleshooting_message()
+
+        internal_debugging = ""
         run_url = os.getenv("GH_RUN_URL")
         if run_url is not None:
-            msg += f"\nRaised by [workflow job]({run_url})"
-        if args.land_checks:
-            msg += get_land_check_troubleshooting_message()
+            # Hide this behind a collapsed bullet since it's not helpful to most devs
+            internal_debugging = (
+                "<details><summary>Details for Dev Infra team</summary>" +
+                f"\nRaised by <a href=\"{run_url}\">workflow job</a>\n" +
+                "</details>"
+            )
+
+        msg = (
+            f"## {title}\n" +
+            f"{exception}\n\n" +
+            f"{troubleshooting}\n\n" +
+            f"{internal_debugging}\n"
+        )
```

Comment:
Nit (use join rather than concatenation
```suggestion
        msg = "\n".join(
            f"## {title}",
            f"{exception}\n",
            f"{troubleshooting}\n",
            f"{internal_debugging}",
        )
```

Reply:
You don't even need to append f-string I think

```
msg = (
    f"## {title}\n"
    f"{exception}\n"
    f"{troubleshooting}\n"
    f"{internal_debugging}"
)
``` 

works as well